ml-algorithms/src/main/java/org/opensearch/ml/engine/encryptor/EncryptorImpl.java

```diff
@@ -102,94 +111,246 @@ public String decrypt(String encryptedText) {
+
+        if (throwable != null) {
+            handleGetDataObjectFailure(throwable, exceptionRef, latch);
+        } else {
+            handleGetDataObjectSuccess(response, tenantId, exceptionRef, latch, context);
+        }
+    }
+
+    private void handleGetDataObjectFailure(Throwable throwable, AtomicReference<Exception> exceptionRef, CountDownLatch latch) {
+        Exception cause = SdkClientUtils.unwrapAndConvertToException(throwable);
+        log.debug("Failed to get ML encryption master key from config index", cause);
+        exceptionRef.set(cause);
+        latch.countDown();
+    }
+
+    private void handleGetDataObjectSuccess(
+        GetDataObjectResponse response,
+        String tenantId,
+        AtomicReference<Exception> exceptionRef,
+        CountDownLatch latch,
+        ThreadContext.StoredContext context
+    ) {
+        try {
+            GetResponse gr = response.parser() == null ? null : GetResponse.fromXContent(response.parser());
+            if (gr != null && gr.isExists()) {
+                this.tenantMasterKeys
+                    .put(Objects.requireNonNullElse(tenantId, DEFAULT_TENANT_ID), (String) response.source().get(MASTER_KEY));
+                log.info("ML encryption master key already initialized, no action needed");
+                latch.countDown();
+            } else {
+                initializeNewMasterKey(tenantId, exceptionRef, latch, context);
             }
-        }, e -> {
-            log.debug("Failed to init ML config index", e);
+        } catch (Exception e) {
+            log.debug("Failed to get ML encryption master key from config index", e);
             exceptionRef.set(e);
             latch.countDown();
-        }));
+        }
+    }
+
+    private void initializeNewMasterKey(
+        String tenantId,
+        AtomicReference<Exception> exceptionRef,
+        CountDownLatch latch,
+        ThreadContext.StoredContext context
+    ) {
+        final String generatedMasterKey = generateMasterKey();
+        sdkClient
+            .putDataObjectAsync(
+                createPutDataObjectRequest(tenantId, generatedMasterKey),
+                client.threadPool().executor("opensearch_ml_general")
+            )
+            .whenComplete(
+                (putDataObjectResponse, throwable1) -> handlePutDataObjectResponse(
+                    tenantId,
+                    context,
+                    putDataObjectResponse,
+                    throwable1,
+                    exceptionRef,
+                    latch,
+                    generatedMasterKey
+                )
+            );
+    }
+
+    private PutDataObjectRequest createPutDataObjectRequest(String tenantId, String generatedMasterKey) {
+        return PutDataObjectRequest
+            .builder()
+            .tenantId(tenantId)
+            .index(ML_CONFIG_INDEX)
+            .id(MASTER_KEY)
+            .overwriteIfExists(false)
```

Comment:
make sure this can work, otherwise will have race condition issue

Reply:
[This PR](https://github.com/opensearch-project/ml-commons/pull/2640) added this functionality. Currently we have unit tests, but we'll also integration tests on this as well. @dbwiddis may be we can have an action item on this as a part of [this issue](https://github.com/opensearch-project/ml-commons/issues/2650)?

Reply:
> [This PR](https://github.com/opensearch-project/ml-commons/pull/2640) added this functionality. Currently we have unit tests, but we'll also integration tests on this as well. @dbwiddis may be we can have an action item on this as a part of [this issue](https://github.com/opensearch-project/ml-commons/issues/2650)?

The PR didn't so much "add" a capability as align the API to the existing capability.  The Unit tests validates that this code produces the same Index Request as the original with `OpType.CREATE`.

If an integ test is missing, it's currently missing on `main` for this class and should be put there separate from this migration.  

is this a gap in current ML Commons integ testing?